smaug/operators/padding_op.h

```diff
@@ -28,42 +28,42 @@ class PaddingOp : public Operator {
 
     /**
      * Set the paddingSize of the Tensor along each dimension.
-     * The paddingSize is orgainized as <dim1_forward, dim1_backward, ...
-     * ,dimk_backward>
+     * The paddingSize is orgainized as <{dim0_begin, dim0_end, dim1_begin,
+     * dim1_end, ... >
      */
-    void setPaddingSize(RepeatedField<google::protobuf::int32> const& val) {
+    void setPaddingSize(const RepeatedField<google::protobuf::int32>& val) {
         paddingSize.assign(val.begin(), val.end());
     }
 
     void setPaddingSize(std::vector<int> const& val) { paddingSize = val; }
 
-    std::vector<int> getPaddingSize() const { return paddingSize; }
+    const std::vector<int>& getPaddingSize() const { return paddingSize; }
 
     void run() override {
-        Tensor* input = getInput(0);
-        Tensor* output = getOutput(0);
+        Tensor* input = getInput(kInput);
+        Tensor* output = getOutput(kOutput);
         int ndims = input->ndims();
-        const std::vector<int> inputDims = input->getShape().dims();
-        const std::vector<int> outputDims = output->getShape().dims();
+        const std::vector<int>& inputDims = input->getShape().dims();
+        const std::vector<int>& outputDims = output->getShape().dims();
         int total_dim = 1;
         for (int i : outputDims) {
             total_dim *= i;
         }
         std::vector<float> vf(total_dim, 0);
         output->fillData(vf.data(), vf.size());
-        std::vector<int> destOrigin, paddingBegin, srcOrigin;
+        std::vector<int> paddingBegin, srcOrigin;
         for (int i = 0; i < ndims; i++) {
-            paddingBegin.push_back(paddingSize[2 * i]);
+            paddingBegin.push_back(paddingSize.at(2 * i));
             srcOrigin.push_back(0);
         }
-        destOrigin = std::vector<int>(paddingBegin);
+        std::vector<int> destOrigin = std::vector<int>(paddingBegin);
         std::vector<int> regionSize = inputDims;
```

Comment:
Same here - no need to make a copy of inputDims, just use it directly. I think you're trying to make it more clear what each vector is representing in the copyTensorRegion call but there's no need since the API documents it very clearly already.